### PR TITLE
Use integer for Npath Complexity

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring

See https://github.com/pdepend/pdepend/issues/843

It should be fine to remove the MathUtil as it assumed bcmul to be installed and was written with a 32bit system in mind. We can now just make the assumption that it's running on a 64bit system, even if wrong this will have far less issues than the old solution for the same case.

### CODE REVIEW

File: src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php [hunk 1]
```diff
@@ -199,19 +197,18 @@ public function visitConditionalExpression(ASTConditionalExpression $node): void
...
-            $npath = MathUtil::add($npath, $cn);
+            $npath += $this->sumComplexity($child);
         }
 
         // Add 2 for the branching per the NPath spec
-        $npath = MathUtil::add($npath, '2');
+        $npath += 2;
 
-        $this->collector = MathUtil::mul($npath, $this->collector);
+        $this->collector *= $npath;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php:211]: If `$this->collector` is `3_037_000_500` (or `46_341` in 32bit) and `$npath` is the same, we are fine with letting face the error:

```
Fatal error: Uncaught TypeError: Cannot assign float to property \PDepend\Metrics\Analyzer\NPathComplexityAnalyzer::$collector of type int
```

Or we want to do our own check before to throw a custom error?

Comment (AJenbo) [reply src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php:211]: I'm ok with that error. If anyone creates an issue I would reply "Please use 64bit builds to analyze huge files"

Comment (ravage84) [reply src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php]: I would prefer a custom exception message, as mentioned in https://github.com/pdepend/pdepend/issues/843#issuecomment-2129721562.

File: src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php [hunk 2]
```diff
@@ -236,11 +233,10 @@ public function visitDoWhileStatement(ASTDoWhileStatement $node): void
         $this->dispatch($node->getChild(0));
         $expr = $this->sumComplexity($node->getChild(1));
 
-        $npath = MathUtil::add($expr, $this->collector);
-        $npath = MathUtil::add($npath, '1');
+        $npath = $expr + $this->collector + 1;
```
Comment (ravage84) [inline src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php]: I would go with the collector always at the beginning:

```suggestion
        $npath = $this->collector + $expr + 1;
```

**Review by ravage84 — COMMENTED** [review]
Generally looks good to me.

As mentioned in the chat, I think `collector` is somewhat ambiguous.

File: src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php (hunk 1 above)
Comment (AJenbo) [reply src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php]: hm in that case should we do this in all places where * or + is used in all of the code base? I would think that requiring 64bit would put the potential numbers well within the safe range of an int.

There are about 50 such operations in the code base, I think we would be one of the first applications written in PHP to guard against 64bit overflow, at least with a clear indication that such a case is likely.

Comment (AJenbo) [reply src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php]: I added 64bit as a requirement to the composer file.

File: src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php (hunk 2 above)
Comment (AJenbo) [reply src/main/php/PDepend/Metrics/Analyzer/NPathComplexityAnalyzer.php]: Done (also the other case)

**Review by kylekatarnls — APPROVED** [review]

**Pull request merged**